File: src/main/java/org/cloudfoundry/credhub/controller/v1/VersionController.java
package org.cloudfoundry.credhub.controller.v1;

import java.util.Map;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import com.google.common.collect.ImmutableMap;
import org.cloudfoundry.credhub.config.VersionProvider;

@RestController
@RequestMapping(produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
public class VersionController {

  private final String credhubVersion;

  @Autowired
  VersionController(
    final VersionProvider versionProvider
  ) {
    super();
    this.credhubVersion = versionProvider.currentVersion();
  }

  @RequestMapping(method = RequestMethod.GET, path = "/version")
  public Map<String, ?> version() {
    return ImmutableMap.of("version", credhubVersion);
  }
}

File: backends/credhub/src/main/kotlin/org/cloudfoundry/credhub/versions/VersionController.kt
package org.cloudfoundry.credhub.versions

import org.cloudfoundry.credhub.utils.VersionProvider
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(path = [VersionController.ENDPOINT], produces = [MediaType.APPLICATION_JSON_UTF8_VALUE])
class VersionController(val versionProvider: VersionProvider) {
    companion object {
        const val ENDPOINT = "/version"
    }

    @RequestMapping(method = [RequestMethod.GET], path = [""])
    fun version(): Map<String, String> {
        return mapOf("version" to versionProvider.currentVersion())
    }
}